### PR TITLE
Ci test binary

### PR DESCRIPTION
**Pull request by brian-js**
[Review on Reviewable](https://reviewable.io/reviews/maidsafe/routing/898)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @canndrew

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: tests/ci.rs [hunk 1]
```diff
@@ -24,23 +32,19 @@ use std::sync::mpsc::{self, Sender, Receiver, TryRecvError};
...
-use node::Node;
-use plain_data::PlainData;
+use routing::Authority;
+use routing::Client;
+use routing::Data;
+use routing::Event;
+use routing::FullId;
+use routing::{RequestContent, ResponseContent, RequestMessage, ResponseMessage};
+use routing::Node;
+use routing::PlainData;
```
Comment (Fraser999) [inline tests/ci.rs]: These `routing` statements could all be a single one.

File: Cargo.toml [hunk 2]
```diff
@@ -40,3 +40,9 @@ test = true
 bench = false
 name = "ci_test"
 test = true
+
+[[test]]
+bench = false
+harness = false
+name = "ci"
+test = true
```
Comment (Fraser999) [inline Cargo.toml]: `test` is true by default.

**Comment by Fraser999** [comment]
Please run rustfmt on this file.

File: Cargo.toml (hunk 2 above)
Comment (brian-js) [inline Cargo.toml]: It's explicitly true now though.

Comment (Fraser999) [inline Cargo.toml]: I see my mistake - I wasn't explicit enough :)  So... could you remove the unnecessary `test = true` lines please?

**Comment by madadam** [comment]
There are two errors in this PR, which were actually introduced in #897, but they sort of cancel each other out so I only noticed them now:

First one is here: https://github.com/brian-js/routing/blob/ci_test_binary/tests/ci.rs#L112, originally, the line was `let interval = Duration::from_millis(100)`. This change makes the `recv_with_timeout` function less useful, because it now always waits the full `timeout` duration (unless there already was a message in the queue at the time of calling it). Before the change, the function only waited until a message was received. So this change makes the tests take longer to finish. 

The second one is here: https://github.com/brian-js/routing/blob/ci_test_binary/tests/ci.rs#L263 and here: https://github.com/brian-js/routing/blob/ci_test_binary/tests/ci.rs#L312 - the comparison operator is reversed. It should be `start + timeout < time::SteadyTime::now()` - so the loop breaks when it takes more than the given timeout, not less :)

My suggestion is to revert the `recv_with_timeout` function to how it was before and then you can remove the timeout checks in request/response to group authority tests completely, because `recv_with_timeout` takes care of it automatically.

**Comment by brian-js** [comment]
Having the comparison operator inverted certainly doesn't help. Cheers @madadam.

**Pull request merged**